core/src/main/java/io/smallrye/stork/impl/CachingServiceDiscovery.java

```diff
@@ -42,11 +42,16 @@ public CachingServiceDiscovery(String refreshPeriod) {
         }
 
         this.lastResults = Collections.emptyList();
-        this.instances = Uni.createFrom().deferred(() -> fetchNewServiceInstances(this.lastResults)
+        Uni<List<ServiceInstance>> least = Uni.createFrom().deferred(() -> fetchNewServiceInstances(this.lastResults)
                 .invoke(l -> this.lastResults = l)
                 .onFailure().invoke(this::handleFetchError)
-                .onFailure().recoverWithItem(this.lastResults))
-                .memoize().atLeast(this.refreshPeriod);
+                .onFailure().recoverWithItem(this.lastResults));
+        this.instances = cache(least);
+    }
+
+    // Overridable by the provider
```

Comment:
I would add a javadoc block explaining:

1. the default behavior + default property
2. it can be overriden